iina/PlaybackHistory.swift

```diff
@@ -63,7 +63,7 @@ class PlaybackHistory: NSObject, NSCoding {
   init(url: URL, duration: Double, name: String? = nil) {
     self.url = url
     self.name = name ?? url.lastPathComponent
-    self.mpvMd5 = Utility.mpvWatchLaterMd5(url.path)
```

Comment:
mpv has a little more logic to calculate file hashes, especially:
- When the file is a network URL,
- Or user specified `--ignore-path-in-watch-later-config`, in which case the md5 is calculated from the filename only.

To stay consistent with mpv, IINA should implement all these logics. It's not implemented yet (all `Utility.mpvWatchLaterMd5` do is returning the md5 of the path), but probably will be in the future. So I suggest still using `Utility.mpvWatchLaterMd5` for better compatibility.

Reply:
Having a Utility type is an anti-pattern. I'm not sure I entirely follow the use case. I guess I'll setup an MD5Hash struct and use that? It will have multiple constructors which will all hash the argument appropriately.

Reply:
If you could merge this so that it doesn't get stale. I'll submit the hash thing very soon.